### PR TITLE
[bitnami/airflow] Chart standardization

### PR DESCRIPTION
**Pull request by juan131**
Signed-off-by: juan131 <juanariza@vmware.com>

**Description of the change**

This PR standardizes the `bitnami/airflow` chart to be in line with the rest of the catalog.

**Benefits**

Standardization.

**Possible drawbacks**

Changes the values structure forcing a major bump, check the README.md for more information.

**Applicable issues**

N/A

**Additional information**

Triggered after https://github.com/bitnami/charts/pull/8827 (major bump in PostgreSQL)

**Checklist**

- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [x] Variables are documented in the values.yaml and added to the README.md using (readme-generator-for-helm)[https://github.com/bitnami-labs/readme-generator-for-helm]
- [x] Title of the PR starts with chart name (e.g. [bitnami/<name_of_the_chart>])

### CODE REVIEW

**Review by migruiz4 — CHANGES_REQUESTED** [review]
Hi, please check my comments below.

In addition, I have noticed the value `*.lifecycleHooks` is missing.

File: bitnami/airflow/templates/_helpers.tpl [hunk 1]
```diff
@@ -274,6 +265,25 @@ Add environment variables to configure database values
...
+            {{- printf "%s" .Values.externalRedis.existingSecretPasswordKey -}}
+        {{- else -}}
+            {{- printf "%s" "redis-password" -}}
+        {{- end -}}
+    {{- else -}}
+        {{- printf "%s" "redis-password" -}}
+    {{- end -}}
+{{- end -}}
+{{- end -}}
+
```
[the remark this answers is not in the document]
Comment (juan131) [reply bitnami/airflow/templates/_helpers.tpl:286]: Yes, it's intentional. The default secret key changed in the latest PostgreSQL major, but Redis isn't affected.

File: bitnami/airflow/templates/metrics/deployment.yaml [hunk 2]
```diff
@@ -66,21 +86,9 @@ spec:
...
-            {{- end }}
-          envFrom:
-            {{- if .Values.metrics.extraEnvVarsCM }}
-            - configMapRef:
-                name: {{ .Values.metrics.extraEnvVarsCM }}
-            {{- end }}
-            {{- if .Values.metrics.extraEnvVarsSecret }}
-            - secretRef:
-                name: {{ .Values.metrics.extraEnvVarsSecret }}
-            {{- end }}
```
[the remark this answers is not in the document]
Comment (juan131) [reply bitnami/airflow/templates/metrics/deployment.yaml:80]: recovered

File: bitnami/airflow/templates/config/secret.yaml [hunk 3]
```diff
@@ -3,34 +3,18 @@ apiVersion: v1
...
-  airflow-secretKey: {{ .Values.auth.secretKey | b64enc | quote }}
-  {{ else if (not .Values.auth.forcePassword) }}
-  airflow-secretKey: {{ randAlphaNum 32 | b64enc | b64enc | quote }}
-  {{ else }}
-  airflow-secretKey: {{ required "An Airflow Secret Key is required!" .Values.auth.secretKey }}
-  {{- end }}
+  airflow-password: {{ include "common.secrets.passwords.manage" (dict "secret" (include "common.names.fullname" .) "key" "airflow-password" "providedValues" (list "auth.password") "context" $) }}
+  # Airflow keys must be base64-encoded, hence we need to pipe to 'b64enc' twice
+  airflow-fernet-key: {{ include "common.secrets.passwords.manage" (dict "secret" (include "common.names.fullname" .) "key" "airflow-fernet-key" "providedValues" (list "auth.fernetKey") "length" 32 "context" $) | b64enc | quote }}
+  airflow-secret-key: {{ include "common.secrets.passwords.manage" (dict "secret" (include "common.names.fullname" .) "key" "airflow-secret-key" "providedValues" (list "auth.secretKey") "length" 32 "context" $) | b64enc | quote }}
```
[the remark this answers is not in the document]
Comment (juan131) [reply bitnami/airflow/templates/config/secret.yaml:19]: Right, it's documented in the Upgrading Notes

File: bitnami/airflow/templates/scheduler/deployment.yaml [hunk 4]
```diff
@@ -113,27 +114,31 @@ spec:
             - secretRef:
                 name: {{ .Values.scheduler.extraEnvVarsSecret }}
             {{- end }}
-          ports:
-            - name: task-logs
-              containerPort: {{ .Values.worker.port }}
```
[the remark this answers is not in the document]
Comment (juan131) [reply bitnami/airflow/templates/scheduler/deployment.yaml:118]: I've been doing some tests and I found no process listening in that port.

File: bitnami/airflow/values.yaml [hunk 5]
```diff
@@ -906,118 +997,407 @@ ldap:
...
-  ## For a full list of possible ingress annotations, please see
-  ## ref: https://github.com/kubernetes/ingress-nginx/blob/master/docs/user-guide/nginx-configuration/annotations.md
+  apiVersion: ""
+  ## @param ingress.hostname Default host for the ingress record
+  ##
+  hostname: airflow.local
+  ## @param ingress.path Default path for the ingress record
+  ## NOTE: You may need to set this to '/*' in order to use this with ALB ingress controllers
+  ##
+  path: /pulse
```
[the remark this answers is not in the document]
Comment (juan131) [reply bitnami/airflow/values.yaml:1069]: Copy-paste error

File: bitnami/airflow/values.yaml [hunk 6]
```diff
@@ -906,118 +997,407 @@ ldap:
...
-      ##
-      tlsSecret: airflow.local-tls
-  ## Key and certificate should start with -----BEGIN CERTIFICATE----- or
-  ## -----BEGIN RSA PRIVATE KEY-----
-  ##
-  ## name should line up with a tlsSecret set further up
-  ## If you're using cert-manager, this is unneeded, as it will create the secret for you if it is not set
+  annotations:
+    # Session affinity is required, see https://kubernetes.github.io/ingress-nginx/examples/affinity/cookie/
+    nginx.ingress.kubernetes.io/affinity: cookie
```
[the remark this answers is not in the document]
Comment (juan131) [reply bitnami/airflow/values.yaml:1080]: Copy-paste error

File: bitnami/airflow/templates/_helpers.tpl (hunk 1 above)
[the remark this answers is not in the document]
Comment (migruiz4) [reply bitnami/airflow/templates/_helpers.tpl:286]: Understood, thanks

**Review by migruiz4 — APPROVED** [review]
LGTM!

File: bitnami/airflow/Chart.yaml [hunk 7]
```diff
@@ -11,12 +11,12 @@ dependencies:
...
     repository: https://charts.bitnami.com/bitnami
-    version: 10.x.x
+    version: 11.x.x
   - condition: redis.enabled
     name: redis
     repository: https://charts.bitnami.com/bitnami
-    version: 15.x.x
-description: Apache Airflow is a tool to express and execute workflows as directed acyclic graphs (DAGs). It includes utilities to schedule tasks, monitor task progress and handle task dependencies.
+    version: 16.x.x
+description: Apache Airflow is a platform to programmatically author, schedule and monitor workflows.
```
[the remark this answers is not in the document]
Comment (migruiz4) [reply bitnami/airflow/Chart.yaml:19]: Updating the majors of Redis and Postresql that have been already standardized.

[the remark this answers is not in the document]
Comment (migruiz4) [reply bitnami/airflow/Chart.yaml:19]: Sorry, you meant the description, right?

**Review by migruiz4 — APPROVED** [review]
LGTM

File: bitnami/airflow/Chart.yaml (hunk 7 above)
[the remark this answers is not in the document]
Comment (juan131) [reply bitnami/airflow/Chart.yaml:19]: I didn't change that... It seems the description was changed in the commit below:
- https://github.com/bitnami/charts/commit/fd78a9b306a82ef0d30418b5b47db45c254cf07b
Let me rebase and update the PR

**Review by alemorcuq — DISMISSED** [review]

**Review by migruiz4 — DISMISSED** [review]
Changes to adopt `global.postgresql` approved

**Pull request merged**